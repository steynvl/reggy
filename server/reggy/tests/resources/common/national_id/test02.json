{
  "type": "National ID",
  "information": {
    "kind": "South African ID number"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
