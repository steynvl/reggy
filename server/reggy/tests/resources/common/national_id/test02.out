Java
Pattern regex = Pattern.compile("\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}");
\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}
Perl
my $regex = /\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}/;
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
POSIX
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
Python
regex = re.compile(r'\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}')
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
JavaScript
const regex = /\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}/;
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
PHP
$regex = '/\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}/';
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
Golang
regex, _ := regexp.Compile("\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}")
\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}
Rust
let re = Regex::new(r"\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}").unwrap();
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
C#
var re = new Regex(@"\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}");
\d{2}[0-1][0-9][0-3]\d\d{4}[0-1]\d{2}
Scala
val re = "\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}".r
\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}
Kotlin
val regex = Regex("\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}")
\\d{2}[0-1][0-9][0-3]\\d\\d{4}[0-1]\\d{2}